driver/src/price_finding/optimization_price_finder.rs

```diff
@@ -223,9 +222,27 @@ impl PriceFinding for OptimisationPriceFinder {
             orders: orders.iter().map(From::from).collect(),
             fee: self.fee.as_ref().map(From::from),
         };
-        let current_time = Utc::now().to_rfc3339();
-        let input_file = format!("instances/instance_{}.json", &current_time);
-        let result_folder = format!("results/instance_{}/", &current_time);
+
+        let now = Utc::now();
+        let batch_id = now.timestamp() / 300;
+        let date = now.format("%Y-%m-%d");
+        // We only need to create the folder for the input, the output
+        // folder is created by the python script
+        let input_folder = format!("instances/{}", &date);
+        create_dir_all(&input_folder)?;
+        let input_file = format!(
+            "{}/instance_{}_{}.json",
+            &input_folder,
+            &batch_id,
+            &now.to_rfc3339()
+        );
+        let result_folder = format!(
+            "results/{}/instance_{}_{}/",
+            &date,
+            &batch_id,
```

Comment:
To have this ordered lexicographically correctly we should 0 pad the batch_id.

Reply:
While that is in general true, I don't think that is needed here as we are currently at batch `5279419` and until batch `10000000` is about 44 years (if my math isn't wrong).

Reply:
It's a valid point, but I wouldn't know how many 0s I should use for padding and it's quite annoying to look at.